src/measure_theory/bochner_integration.lean

```diff
@@ -1666,4 +1670,98 @@ attribute [integral_simps] integral_neg integral_smul L1.integral_add L1.integra
 
 attribute [irreducible] integral L1.integral
 
+section integral_trim
+
+variables {H β γ : Type*} [normed_group H] [measurable_space H]
+  {m m0 : measurable_space β} {μ : measure β}
+
+/-- Simple function seen as simple function of a larger `measurable_space`. -/
+def simple_func.to_larger_space (hm : m ≤ m0) (f : @simple_func β m γ) : simple_func β γ :=
+⟨@simple_func.to_fun β m γ f, λ x, hm _ (@simple_func.measurable_set_fiber β γ m f x),
+  @simple_func.finite_range β γ m f⟩
+
+lemma simple_func.coe_to_larger_space_eq (hm : m ≤ m0) (f : @simple_func β m γ) :
+  ⇑(f.to_larger_space hm) = f :=
+rfl
+
+lemma integral_simple_func_larger_space (hm : m ≤ m0) (f : @simple_func β m F)
+  (hf_int : integrable f μ) :
+  ∫ x, f x ∂μ = ∑ x in (@simple_func.range β F m f), (ennreal.to_real (μ (f ⁻¹' {x}))) • x :=
+begin
+  simp_rw ← f.coe_to_larger_space_eq hm,
+  have hf_int : integrable (f.to_larger_space hm) μ, by rwa simple_func.coe_to_larger_space_eq,
+  rw simple_func.integral_eq_sum _ hf_int,
+  congr,
+end
+
+lemma integral_trim_simple_func (hm : m ≤ m0) (f : @simple_func β m F) (hf_int : integrable f μ) :
+  ∫ x, f x ∂μ = @integral β F m _ _ _ _ _ _ (μ.trim hm) f :=
+begin
+  have hf : @measurable _ _ m _ f, from @simple_func.measurable β F m _ f,
+  have hf_int_m := hf_int.trim hm hf,
+  rw [integral_simple_func_larger_space le_rfl f hf_int_m,
+    integral_simple_func_larger_space hm f hf_int],
+  congr,
+  ext1 x,
+  congr,
+  exact (trim_measurable_set_eq hm (@simple_func.measurable_set_fiber β F m f x)).symm,
+end
+
+lemma integral_trim (hm : m ≤ m0) {f : β → F} (hf : @measurable β F m _ f)
+  (hf_int : integrable f μ) :
```

Comment:
Is the integrability assumption really necessary here? If f is not integrable with respect to the big sigma-algebra, it won't either with respect to the small sigma-algebra, and both integrals will vanish, right?

Reply:
Thanks for spotting this! Indeed I could remove the integrability assumption, here and in the two other results.